src/chainbin/nnet3-chain-combine.cc

```diff
@@ -19,7 +20,66 @@
 
 #include "base/kaldi-common.h"
 #include "util/common-utils.h"
-#include "nnet3/nnet-chain-combine.h"
+#include "nnet3/nnet-utils.h"
+#include "nnet3/nnet-compute.h"
+#include "nnet3/nnet-chain-diagnostics.h"
+
+
+namespace kaldi {
+namespace nnet3 {
+
+// Computes the objective of the moving average of nnet on egs. If either of
```

Comment:
Documentation for code should document what the code does, not how it's being used; and it should
be precise.  So here, it should say something like:
`Computes the objective function for the examples in 'egs' given the model in 'nnet'.`
(and rename moving_average_nnet to 'nnet').
This code doesn't need to care how 'moving_average_nnet' was obtained.
And this documentation should say what the return value is.